src/modules/simulator/accelsim/accelsim.cpp

```diff
@@ -318,65 +315,45 @@ ACCELSIM::~ACCELSIM()
 int
 ACCELSIM::init()
 {
-	int ret = -1;
-
-	struct mag_report mrp = {};
-	sensor_accel_s arp = {};
-
 	/* do SIM init first */
 	if (VirtDevObj::init() != 0) {
 		PX4_WARN("SIM init failed");
-		goto out;
+		return PX4_ERROR;
```

Comment:
Can you return the error code from `VirtDevObj::init()`?

Reply:
Done and pushed up.  Thanks!